datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/mappers/DatasetUpdateInputMapper.java

```diff
@@ -0,0 +1,96 @@
+package com.linkedin.datahub.graphql.types.mappers;
+
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.InstitutionalMemory;
+import com.linkedin.common.InstitutionalMemoryMetadata;
+import com.linkedin.common.InstitutionalMemoryMetadataArray;
+import com.linkedin.common.Owner;
+import com.linkedin.common.OwnerArray;
+import com.linkedin.common.Ownership;
+import com.linkedin.common.OwnershipSource;
+import com.linkedin.common.OwnershipSourceType;
+import com.linkedin.common.OwnershipType;
+import com.linkedin.common.url.Url;
+import com.linkedin.common.urn.CorpuserUrn;
+import com.linkedin.datahub.graphql.generated.DatasetUpdateInput;
+import com.linkedin.datahub.graphql.generated.InstitutionalMemoryMetadataUpdate;
+import com.linkedin.datahub.graphql.generated.OwnerUpdate;
+import com.linkedin.dataset.Dataset;
+import com.linkedin.dataset.DatasetDeprecation;
+
+import javax.annotation.Nonnull;
+import java.net.URISyntaxException;
+import java.util.stream.Collectors;
+
+public class DatasetUpdateInputMapper implements ModelMapper<DatasetUpdateInput, Dataset> {
+
+    public static final DatasetUpdateInputMapper INSTANCE = new DatasetUpdateInputMapper();
+
+    public static Dataset map(@Nonnull final DatasetUpdateInput datasetUpdateInput) {
+        return INSTANCE.apply(datasetUpdateInput);
+    }
+
+    @Override
+    public Dataset apply(@Nonnull final DatasetUpdateInput datasetUpdateInput) {
+        final Dataset result = new Dataset();
+
+        if (datasetUpdateInput.getOwnership() != null) {
+            final Ownership ownership = new Ownership();
+            ownership.setOwners(new OwnerArray(datasetUpdateInput.getOwnership().getOwners()
+                    .stream()
+                    .map(this::mapOwner)
+                    .collect(Collectors.toList())));
+            result.setOwnership(ownership);
+        }
+
+        if (datasetUpdateInput.getDeprecation() != null) {
+            final DatasetDeprecation deprecation = new DatasetDeprecation();
+            deprecation.setDeprecated(datasetUpdateInput.getDeprecation().getDeprecated());
+            deprecation.setDecommissionTime(datasetUpdateInput.getDeprecation().getDecommissionTime());
+            deprecation.setNote(datasetUpdateInput.getDeprecation().getNote());
+            result.setDeprecation(deprecation);
+        }
```

Comment:
why isn't there a mapDeprecation method on this class?

Reply:
I only added "mapX" methods for mapping individual items in a complex array (elements, owners). Let's keep for now... If this class grows more complicated we can consider a refactor!